oak-segment-aws/src/main/java/org/apache/jackrabbit/oak/segment/aws/AwsGCJournalFile.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.segment.aws;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.jackrabbit.oak.segment.spi.persistence.GCJournalFile;
+
+public class AwsGCJournalFile implements GCJournalFile {
```

Comment:
I think this is not correctly implemented as it uses an `AwsJournalFile` under the hood
// the problem is that the entries need not be in reverse order as imposed by `JournalFileReader` interface.

Reply:
it does do the right thing... though I agree... using JournalFileReader makes this confusing... I've removed AwsAppendableFile.java altogether